radix-engine-stores/src/interface.rs

```diff
@@ -158,10 +158,14 @@ pub trait SubstateStore {
 }
 
 #[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor)]
-pub struct DatabaseUpdates {
-    pub database_updates: IndexMap<Vec<u8>, IndexMap<Vec<u8>, DatabaseUpdate>>,
+pub struct StateUpdates {
+    pub database_updates: DatabaseUpdates,
```

Comment:
I don't think we should be returning both. This doubles the size of our receipt for no reason.

Instead, just exposing SystemUpdates is fine and we can map to the Database keys super easily in the client

Reply:
This can be changed in a follow-up PR though

Reply:
Maybe, it feels like a weird abstraction to me but lets chat about it when we get to it.